src/test/java/io/cdap/plugin/servicenow/source/ServiceNowSourceTest.java

```diff
@@ -16,108 +16,234 @@
+      "            \"title\": \"\",\n" +
+      "            \"sys_class_name\": \"sys_user\",\n" +
+      "            \"sys_id\": \"005d500b536073005e0addeeff7b12f4\",\n" +
+      "            \"internal_integration_user\": \"false\",\n" +
+      "            \"ldap_server\": \"\",\n" +
+      "            \"mobile_phone\": \"\",\n" +
+      "            \"street\": \"\",\n" +
+      "            \"company\": \"\",\n" +
+      "            \"department\": \"\",\n" +
+      "            \"first_name\": \"survey\",\n" +
+      "            \"email\": \"survey.user@email.com\",\n" +
+      "            \"introduction\": \"\",\n" +
+      "            \"preferred_language\": \"\",\n" +
+      "            \"manager\": \"\",\n" +
+      "            \"sys_mod_count\": \"1\",\n" +
+      "            \"last_name\": \"user\",\n" +
+      "            \"photo\": \"\",\n" +
+      "            \"avatar\": \"\",\n" +
+      "            \"middle_name\": \"\",\n" +
+      "            \"sys_tags\": \"\",\n" +
+      "            \"time_zone\": \"\",\n" +
+      "            \"schedule\": \"\",\n" +
+      "            \"date_format\": \"\",\n" +
+      "            \"location\": \"\"\n" +
+      "        }\n" +
+      "    ]\n" +
+      "}";
+    RestAPIResponse restAPIResponse = new RestAPIResponse(httpStatus, headers, responseBody);
+    Mockito.when(restApi.executeGet(Mockito.any())).thenReturn(restAPIResponse);
+    Mockito.when(restApi.parseResponseToResultListOfMap(restAPIResponse.getResponseBody())).thenReturn(result);
+    OAuthClient oAuthClient = Mockito.mock(OAuthClient.class);
+    PowerMockito.whenNew(OAuthClient.class).
+      withArguments(Mockito.any(URLConnectionClient.class)).thenReturn(oAuthClient);
+    OAuthJSONAccessTokenResponse accessTokenResponse = Mockito.mock(OAuthJSONAccessTokenResponse.class);
+    Mockito.when(oAuthClient.accessToken(Mockito.any(), Mockito.anyString(), Mockito.any(Class.class))).
+      thenReturn(accessTokenResponse);
+    Mockito.when(accessTokenResponse.getAccessToken()).thenReturn("token");
+    RestAPIResponse response = Mockito.spy(restAPIResponse);
+    CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+    HttpClientBuilder httpClientBuilder = Mockito.mock(HttpClientBuilder.class);
+    PowerMockito.mockStatic(HttpClientBuilder.class);
+    PowerMockito.mockStatic(RestAPIResponse.class);
+    PowerMockito.when(HttpClientBuilder.create()).thenReturn(httpClientBuilder);
+    Mockito.when(httpClientBuilder.build()).thenReturn(httpClient);
+    CloseableHttpResponse httpResponse = Mockito.mock(CloseableHttpResponse.class);
+    Mockito.when(httpClient.execute(Mockito.any())).thenReturn(httpResponse);
+    PowerMockito.when(RestAPIResponse.parse(ArgumentMatchers.any(), ArgumentMatchers.anyString())).
+      thenReturn(response);
+    serviceNowSource.configurePipeline(mockPipelineConfigurer);
+    Assert.assertEquals(0, mockFailureCollector.getValidationFailures().size());
```

Comment:
Try to validate something positive, validating failure size is not a good practice.

Reply:
Added another assertion for checking `Schema.Type` c0e6a0f